app/controllers/memberships_controller.rb

```diff
@@ -7,13 +7,7 @@ def create
       CreateMembership.call(@group, current_user)
       redirect_back(fallback_location: group_path(@group))
     else
-      membership = @group.member?(current_user)
-      if membership && membership.can_request_unarchivation?
-        request_reactivation(membership)
-        redirect_back(fallback_location: group_path(@group))
-      else
-        forbidden_page
-      end
+      handle_unarchivation_request
```

Comment:
I moved the logic to a private method, to reduce complexity in the `create` method.